### ISSUE TITLE
Actions do not support returning scalar types

### ISSUE THREAD

**Issue by samuela** [hasura/graphql-engine#7805]
### Version Information

Server Version: v2.0.10
CLI Version (for CLI related issue): n/a

### Environment

macOS 11.6

### What is the expected behaviour?

Creating an Action like
```gql
type Mutation {
  StartCuddlefishSession2(github_access_token: String!): String!
}
```
to work.

### Keywords

Actions, graphql, scalar types, custom types

### What is the current behaviour?

I'm getting an error:
```
{
    "internal": [
        {
            "definition": {
                "definition": {
                    "kind": "synchronous",
                    "output_type": "String!",
                    "arguments": [
                        {
                            "name": "github_access_token",
                            "type": "String!",
                            "description": null
                        }
                    ],
                    "headers": [
                        {
                            "name": "x-api-secret",
                            "value_from_env": "API_SECRET"
                        }
                    ],
                    "handler": "http://{{NEXT_HOST}}:{{NEXT_PORT}}/api/start_cuddlefish_session",
                    "type": "mutation",
                    "timeout": 30,
                    "forward_client_headers": false
                },
                "name": "StartCuddlefishSession2",
                "comment": "Start a new Cuddlefish session provided a GitHub auth token"
            },
            "reason": "Inconsistent object: in action \"StartCuddlefishSession2\"; the type: \"String\" is not an object type defined in custom types",
            "name": "action StartCuddlefishSession2",
            "type": "action"
        }
    ],
    "path": "$.args[1].args",
    "error": "Inconsistent object: in action \"StartCuddlefishSession2\"; the type: \"String\" is not an object type defined in custom types",
    "code": "invalid-configuration"
}
```

### How to reproduce the issue?

1. Try to create an action with a mutation type, returning a base scalar type like `String`
2. Empty out the "New types definition" text box.
3. Click "Create"

### Screenshots or Screencast

<img width="1259" alt="Screen Shot 2021-11-11 at 7 34 00 PM" src="https://user-images.githubusercontent.com/226872/141404585-25945a34-e76a-4db6-a889-0152b23a9f56.png">

### Please provide any traces or logs that could help here.

n/a

### Any possible solutions?

Having to create custom types for every action

### Can you identify the location in the source code where the problem exists?

n/a

### If the bug is confirmed, would you be willing to submit a PR?

yes

**Comment by sytabaresa** [comment]
makes sense: if an action can have a null response (see #4405 ), also can respond with a string like "Accepted" or "OK" or another scalar type like a counter number etc.

with the actual version (2.1.0 and 2.1.1) continue the issue

**Comment by ivelin** [comment]
Here is a related scenario to consider 

https://discord.com/channels/407792526867693568/535727661167673364/933298267737964564

**Comment by sordina** [comment]
@0x777 is there any issue with implementing this feature?

**Comment by pranshi06** [comment]
Closed via 582d78110fb8b7e1e30936bcc6ebc3cdbad59d39
